cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -374,6 +386,16 @@ impl fmt::Display for AluRmiROpcode {
     }
 }
 
+impl AluRmiROpcode {
+    /// Is this a special-cased 8-bit ALU op?
+    pub fn is_8bit(self) -> bool {
+        match self {
+            AluRmiROpcode::And8 | AluRmiROpcode::Or8 => true,
+            _ => false,
+        }
+    }
+}
```

Comment:
It seems preferable to me to just have `AluRmiROpcode::And` and avoid this special-casing if possible. Do you agree? If yes, I wonder if there is a way to standardize register sizes throughout `lower.rs` using `OperandSize` and trying to clarify/simplify all of the special logic related to variables like `is64`, `is_64`, `is_8bit`, etc.

Reply:
In general I would agree, except that we don't use 8-bit ops elsewhere for CLIF-level 8-bit instructions so I went for the limited-in-scope approach here. I'm totally open to a future refactor that centralizes `OperandSize` for all instructions, though -- that would remove a lot of the ad-hoc conditionals, as you note!